signing.go

```diff
@@ -0,0 +1,13 @@
+package notation
+
+import (
+	"context"
+
+	oci "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// SigningService provides signature signing and verification services.
+type SigningService interface {
+	Sign(ctx context.Context, desc oci.Descriptor, references ...string) ([]byte, error)
+	Verify(ctx context.Context, desc oci.Descriptor, signature []byte) ([]string, error)

```

Comment:
Signing service doesnt need to know about Context, Descriptors or references. For signing service should only need signingIdentity and data that needs to be signed.

Reply:
This interface will be refined.

- `Descriptor` is required since it is the data to be signed.
- `ctx` is used for remote signing or remote verification.

Reply:
Yes Descriptor is required, I was thinking whether we should pass descriptor as Type or plain byte buffer since signer doesnt need to interpret the data. Also, passing data as bytebuffer avoid json canonicalization issue(For a given byte buffer a given private key will generate same signature).

Yes Context is required, I'm still ramping up on Go.

We should also allow user to pass custom signed attributes.